enterprise/internal/own/background/recent_contributors.go

```diff
@@ -73,3 +88,36 @@ func (r *recentContributorsIndexer) indexRepo(ctx context.Context, repoId api.Re
 	commitCounter.Add(float64(len(commitLog)))
 	return nil
 }
+
+func isSubRepoPermsRepo(ctx context.Context, repoID api.RepoID, cache rcache.Cache, checker authz.SubRepoPermissionChecker) (isSubRepoPermsRepo bool, err error) {
+	cacheKey := strconv.Itoa(int(repoID))
+	// Look for the repo in cache to see if we have seen it before instead of hitting the DB.
+	val, ok := cache.Get(cacheKey)
+	if ok {
+		var isSubRepoPermsRepo bool
+		if err := json.Unmarshal(val, &isSubRepoPermsRepo); err != nil {
+			return false, err
+		}
+		return isSubRepoPermsRepo, nil
+	}
+
+	// No entry in cache, so we need to look up whether this is a sub-repo perms repo in the DB.
+	ok, err = authz.SubRepoEnabledForRepoID(ctx, checker, repoID)
+	if err != nil {
+		return false, err
+	}
+	if ok {
+		b, err := json.Marshal(true)
+		if err != nil {
+			return false, err
+		}
+		cache.Set(cacheKey, b)
+		return true, nil
+	}
+	b, err := json.Marshal(false)
+	if err != nil {
+		return false, err
+	}
+	cache.Set(cacheKey, b)
+	return false, nil
```

Comment:
basically, we can rewrite it as (I consider renaming `ok`->`isSubRepoPermsRepo`):
right?
```suggestion
	b, err := json.Marshal(isSubRepoPermsRepo)
	if err != nil {
		return false, err
	}
	cache.Set(cacheKey, b)
	return isSubRepoPermsRepo, nil
```

Reply:
Great suggestions, thanks Sash!